sdv/constraints/utils.py

```diff
@@ -104,3 +104,59 @@ def sigmoid(data, low, high):
     data = data * (high - low) + low
 
     return data
+
+
+def check_nans_row(row):
+    """Check for NaNs in a pandas row.
+
+    Outputs a concatenated string of the column names with NaNs.
+
+    Args:
+        row (pandas.Series):
+            A pandas row.
+    """
+    columns_with_nans = []
+    for column, value in row.items():
+        if pd.isna(value):
+            columns_with_nans.append(column)
+
+    if columns_with_nans:
+        return ', '.join(columns_with_nans)
+    else:
+        return 'None'
+
+
+def add_nans_column(table_data, list_column_names):
+    """Add a categorical column to the table_data indicating where NaNs are.
+
+    Args:
+        table_data (pandas.DataFrame):
+            The table data.
+        list_column_names (list):
+            The list of column names to check for NaNs.
+    """
+    nan_column_name = '#'.join(list_column_names) + '.nan_component'
+    nan_column = table_data[list_column_names].apply(check_nans_row, axis=1)
+    if not (nan_column == 'None').all():
+        table_data[nan_column_name] = nan_column
+        return True
+    else:
+        return False
```

Comment:
In general I think functions should try to limit themselves to doing one thing. Here we are adding the nan column to the data only in certain cases, and return a boolean indicating whether or not we added it. Instead, we can have this function just return the column and if the nan_column is all `None`, we can just return `None`. Then the caller of this method knows to either add the column if it exists, or skip it if it's none.